packages/eslint-plugin/src/rules/no-magic-numbers.ts

```diff
@@ -129,6 +136,40 @@ export default createRule<Options, MessageIds>({
   },
 });
 
+/**
+ * Convert the value to bigint if it's a string. Otherwise, return the value as-is.
+ * @param value The value to normalize.
+ * @returns The normalized value.
+ */
+function normalizeIgnoreValue(
+  value: bigint | number | string,
+): bigint | number {
+  if (typeof value === 'string') {
+    return BigInt(value.slice(0, -1));
+  }
+
+  return value;
+}
+
+/**
+ * Converts the node to its numeric value, handling prefixed numbers (-1 / +1)
+ * @param node the node to normalize.
+ */
+function normalizeLiteralValue(
+  node: TSESTree.BigIntLiteral | TSESTree.NumberLiteral,
+): bigint | number {
+  if (
+    node.parent.type === AST_NODE_TYPES.UnaryExpression &&
+    ['-', '+'].includes(node.parent.operator)
+  ) {
+    if (node.parent.operator === '-') {
+      return -(node.value ?? BigInt(node.raw));
+    }
+  }
+
+  return node.value ?? BigInt(node.raw);
```

Comment:
[Testing] This is reported as only partially covered in the Codecov report: https://app.codecov.io/gh/typescript-eslint/typescript-eslint/pull/9167?src=pr&el=tree&filepath=packages%2Feslint-plugin%2Fsrc%2Frules%2Fno-magic-numbers.ts&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=typescript-eslint#diff-cGFja2FnZXMvZXNsaW50LXBsdWdpbi9zcmMvcnVsZXMvbm8tbWFnaWMtbnVtYmVycy50cw==. Same with the `return -(...);` a few lines earlier.

In fact, removing the two `?? BigInt(node.raw)`s results in no test failures:

```suggestion
  return node.value;
```

...and this function is only ever called on `node`s that have a defined` node.value`, per the `if (typeof ...)` checks in the `Literal(node): void { ... }`. My advice would be to either pass in `node.value` directly _or_ change the type of the `node` parameter to mention that it has a non-`null` `value`.